components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp

```diff
@@ -126,88 +126,70 @@
         }
 
         if (Boolean.parseBoolean(request.getParameter(SAMLSSOUIConstants.ENABLE_ATTRIBUTE_PROFILE))) {
-
             String claimsCountParameter = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyCounter");
             if (claimsCountParameter != null && !"".equals(claimsCountParameter)) {
-                try {
-                    int claimsCount = Integer.parseInt(claimsCountParameter);
-                    for (int i = 0; i < claimsCount; i++) {
-                        String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
-                        if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
-                            String[] currentClaims = serviceProviderDTO.getRequestedClaims();
-                            boolean isClaimAlreadyAdded = false;
-                            for (String currentClaim : currentClaims) {
-                                if (claim.equals(currentClaim)) {
-                                    isClaimAlreadyAdded = true;
-                                    break;
-                                }
-                            }
-
-                            if (!isClaimAlreadyAdded) {
-                                serviceProviderDTO.addRequestedClaims(claim);
+                int claimsCount = Integer.parseInt(claimsCountParameter);
+                for (int i = 0; i < claimsCount; i++) {
+                    String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
+                    if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
+                        String[] currentClaims = serviceProviderDTO.getRequestedClaims();
+                        boolean isClaimAlreadyAdded = false;
+                        for (String currentClaim : currentClaims) {
```

Comment:
This used only for this particuler segment only. Also when getting from DTO service it should be an array. IMHO, If it's going to convert again, there will be another cost assosiate with it.